From 0778686065ef3007b50b84278ab5f58271f9a086 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 29 Aug 2021 08:02:37 -0700
Subject: [PATCH] Avoid segfaults in NvEglRenderer overlay text display

- try 'fixed' font as a backup for '9x15bold'
- make sure a font was loaded before trying to render the overlay

Signed-off-by: Matt Madison <matt@madison.systems>
---
 .../samples/common/classes/NvEglRenderer.cpp                  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/samples/common/classes/NvEglRenderer.cpp b/samples/common/classes/NvEglRenderer.cpp
index cc7cfef..b5520cd 100644
--- a/samples/common/classes/NvEglRenderer.cpp
+++ b/samples/common/classes/NvEglRenderer.cpp
@@ -134,6 +134,8 @@ NvEglRenderer::NvEglRenderer(const char *name, uint32_t width, uint32_t height,
     XSetForeground(x_display, gc,
                 WhitePixel(x_display, DefaultScreen(x_display)) );
     fontinfo = XLoadQueryFont(x_display, "9x15bold");
+    if (fontinfo == NULL)
+        fontinfo = XLoadQueryFont(x_display, "fixed");
 
     pthread_mutex_lock(&render_lock);
     pthread_create(&render_thread, NULL, renderThread, this);
@@ -452,7 +454,7 @@ NvEglRenderer::renderInternal()
     }
     NvDestroyEGLImage(egl_display, hEglImage);
 
-    if (strlen(overlay_str) != 0)
+    if (fontinfo != NULL && strlen(overlay_str) != 0)
     {
         XSetForeground(x_display, gc,
                         BlackPixel(x_display, DefaultScreen(x_display)));
